src/nvim/edit.c

```diff
@@ -7475,17 +7475,23 @@ static void ins_shift(int c, int lastc)
 
 static void ins_del(void)
 {
-  int temp;
-
-  if (stop_arrow() == FAIL)
+  if (stop_arrow() == FAIL) {
     return;
-  if (gchar_cursor() == NUL) {          /* delete newline */
-    temp = curwin->w_cursor.col;
+  }
+  if (gchar_cursor() == NUL) {          // delete newline
+    const int temp = curwin->w_cursor.col;
```

Comment:
This may add friction later if a vim patch mutates the variable. That's another reason why we generally don't do this.

Reply:
The main intent is to `int temp = ...` on the same line. `const` is a bonus.

I keep doing these things because there's too many variables at the top level when they shouldn't be and some functions are very long, sometimes with `#ifdef`, such as patch 8.0.0685. I would expect loop variables to be local to its `for` loop scope but it rarely happens. There are helpers functions but they don't have `const` so I still have to go through each line when debugging. `const` is half-baked but it's preferable to note as many invariants as possible, especially on cleanup and if `goto` is used, like 8.0.0493.

I like to `const` the pointer contents as well but it's hard to add it on a non-constant pointer; it could be passed to another variable or its address could be passed to a function via `&`. Think of local `const` as in-between step so that it's safer to `const` all the way across callers.